elastic/kibana
# [Snapshot Restore] Fix validation schemas

Fixes: https://github.com/elastic/kibana/issues/139896

## Summary

Adds the missing two repository options to the validation schema and I also introduced the `allow unknown settings` to the `s3`, `azure`, `fs` and `gcs` setting schemas. This config for allowing unkown settings was already present for `hds` repos and will help us prevent these kind of errors in the future. 

Note: the `awsAccount` setting is present in all repo configs, not just for s3.

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial` and kibana with `yarn start`
2. Go to console and run
```
PUT  _snapshot/test?verify=false
{
    "type": "s3",
    "settings": {
      "bucket": "test-bucket",
      "client": "test-client",
      "base_path": "test-path",
      "path_style_access": true
    }
  }
```
3. Navigate to `Stack Management` -> `Snapshot and Restore`, edit the existing repo and save it again without making any changes.
4. Verify that:
    * It saves correctly without any errors
